src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryToolKit.csproj

```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <RuntimeVersion></RuntimeVersion>
+    <OutputType>Exe</OutputType>
+  </PropertyGroup>
+
+  <Target Name="CheckRuntimeVersion" BeforeTargets="CoreCompile">
+    <Error Condition="'$(RuntimeVersion)' == ''" Text="RuntimeVersion is not set. Please consider passing as an MSBuild Property" />
+  </Target>
```

Comment:
Setting this property affects Restore as well so this should probably be an initial target instead. https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-specify-which-target-to-build-first?view=vs-2022#use-the-initialtargets-attribute